### DISCUSSION TITLE
select_prev_item auto calls confirm on emmet ls snippets

### DISCUSSION THREAD

**Discussion by GZLiew** [hrsh7th/nvim-cmp#1777] [Q&A]
```lua
cmp.setup({
  snippet = {
    expand = function(args)
      vim.fn["vsnip#anonymous"](args.body)
    end,
  },
  mapping = {
		['<C-n>'] = function(fallback)
        if cmp.visible() then
          cmp.select_next_item()
        else
          fallback()
        end
      end,
		['<C-p>'] = function(fallback)
        if cmp.visible() then
          cmp.select_prev_item()
        else
          fallback()
        end
      end,
    ['<C-y>'] = cmp.config.disable,
    ['<C-Space>'] = cmp.mapping(cmp.mapping.complete(), { 'i', 'c', 's', 'v' }),
    ['<C-f>'] = cmp.mapping.confirm({ select = true }),
    ['<Tab>'] = cmp.mapping.confirm({ select = true }),
    ['<CR>'] = cmp.mapping.confirm({ select = true }),
  },
  sources = cmp.config.sources({
    { name = 'vsnip' },
    { name = 'nvim_lsp' },
    { name = 'buffer' },
  })
})
```

when i use `select_next_item` to jump to the item it would let me confirm

but select_prev_item auto calls confirm

is there something wrong with my config?

https://github.com/hrsh7th/nvim-cmp/assets/43195293/1155b5f0-0366-4e73-8ac9-9e391b721634

**Comment by LAST7** [comment]
Same issue here. Both select_next_item/select_prev_item auto calls confirm on emmet-language-server here. The issue begins to show up recently (like a couple of weeks ago).

Other lsp seem to work fine, only emmet is having this issue.

**Comment by LAST7** [comment]
Okay seems that I have found a way to fix this. Here's the solution:

put `{behavior = cmp.SelectBehavior.Select }` as a parameter to the `select_next_item` function.

for example:

```
["<Tab>"] = cmp.mapping(function(fallback)
    if cmp.visible() then
        cmp.select_next_item({ behavior = cmp.SelectBehavior.Select })
    else
        fallback()
    end
```

same for `<S-Tab>`.
